esp/bindings/http/client/httpclient.cpp

```diff
@@ -754,7 +755,8 @@ int CHttpClient::postRequest(ISoapMessage &req, ISoapMessage& resp)
 IHttpClientContext* getHttpClientContext()
 {
     static Owned<CHttpClientContext> theHttpClientContext;
-
+    static CriticalSection httpCrit;
```

Comment:
more efficient to have this outside the function.  If you're being really clever/modern you could say

static Owned<CHttpClientContext> theHttpClientContext = new CHqttpClientContext();

which is threadsafe on modern compilers.